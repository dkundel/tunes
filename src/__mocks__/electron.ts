import EventEmitter = require('events');

export class MockIpcRenderer extends EventEmitter {}

export const _ipcRendererEmitter = new MockIpcRenderer();

export const ipcRenderer = {
  send: jest.fn(),
  on: jest.fn((evt: string, callback: Function) =>
    _ipcRendererEmitter.on(evt, callback)
  )
};

export const shell = {
  openExternal: jest.fn()
};
