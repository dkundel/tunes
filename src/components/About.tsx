import * as React from 'react';
import styled from 'styled-components';
import {  } from 'polished';
import { remote } from 'electron';

import { H1, P } from '../styles/components';
import ExternalLink from './ExternalLink';
import MaterialIcon from './MaterialIcon';

const Container = styled.div`
  text-align: center;
  padding: 40px;
`;

const Logo = styled(H1)`
  font-size: 50px;
  margin-bottom: 0;
  margin-top: 0;
`;

const AppDescription = styled(P)`
  text-align: justify;
`;

const InfoTable = styled.div`
  text-align: left;
`;

const InfoTableRow = styled.div`
  display: flex;
`;

const InfoTableEntry = styled.div`
  flex: 1;
`;

const InfoTableLabel = styled(InfoTableEntry)`
  font-weight: 600;
  flex: initial;
  margin-right: 20px;
`;

const CloseButton = styled.button`
  margin: 20px;
  padding: 10px;
  background: none;
  cursor: pointer;
  border: none;
  outline: none;

  &:hover, &:focus {
    color: var(--colorPrimary);
  }
`

export default class About extends React.Component<any, any> {
  closeDialog() {
    remote.getCurrentWindow().close();
  }

  render() {
    return (
      <Container>
        <Logo>ðŸŽµ</Logo>
        <H1>About Tunes</H1>
        <AppDescription>
          This application allows you to play YouTube playlists in a small
          player right in your menubar. It is powered by the YouTube iFrame API
          and therefore might have limitations on which songs can be played. It
          will automatically skip blocked videos.
        </AppDescription>
        <AppDescription>
          The application is developed open-source. You can inspect the source
          on its GitHub repository below.
        </AppDescription>
        <InfoTable>
          <InfoTableRow>
            <InfoTableLabel>Version</InfoTableLabel>
            <InfoTableEntry>{remote.app.getVersion()}</InfoTableEntry>
          </InfoTableRow>
          <InfoTableRow>
            <InfoTableLabel>Developer</InfoTableLabel>
            <InfoTableEntry>Dominik Kundel</InfoTableEntry>
          </InfoTableRow>
          <InfoTableRow>
            <InfoTableLabel>Source Code</InfoTableLabel>
            <InfoTableEntry>
              <ExternalLink
                href="https://github.com/dkundel/tunes"
                target="_blank"
              >
                https://github.com/dkundel/tunes
              </ExternalLink>
            </InfoTableEntry>
          </InfoTableRow>
        </InfoTable>
        <CloseButton onClick={() => this.closeDialog()}>
          <MaterialIcon icon="close"/>
        </CloseButton>
      </Container>
    );
  }
}
