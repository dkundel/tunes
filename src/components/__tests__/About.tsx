jest.mock('electron');

import * as React from 'react';
import About from '../About';
import { shallow } from 'enzyme';
import * as renderer from 'react-test-renderer';

import { remote } from 'electron';

describe('test About component', () => {
  test('verify snapshot', () => {
    const component = renderer.create(<About />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('closes window on button press', () => {
    const component = shallow(<About />);
    component.find('#closeAbout').simulate('click');
    expect(remote.getCurrentWindow().close).toHaveBeenCalledTimes(1);
  });
});
