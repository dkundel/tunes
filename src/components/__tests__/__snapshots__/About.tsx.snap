// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test About component verify snapshot 1`] = `
<div
  className="sc-htpNat dVjawc"
>
  <h1
    className="sc-bxivhb cJfdTg sc-bdVaJa eMtwIx"
  >
    ðŸŽµ
  </h1>
  <h1
    className="sc-bdVaJa eMtwIx"
  >
    About Tunes
  </h1>
  <p
    className="sc-ifAKCX qycJf sc-bwzfXH bGgvvP"
  >
    This application allows you to play YouTube playlists in a small player right in your menubar. It is powered by the YouTube iFrame API and therefore might have limitations on which songs can be played. It will automatically skip blocked videos.
  </p>
  <p
    className="sc-ifAKCX qycJf sc-bwzfXH bGgvvP"
  >
    The application is developed open-source. You can inspect the source on its GitHub repository below.
  </p>
  <div
    className="sc-EHOje kLbaUi"
  >
    <div
      className="sc-bZQynM kzlsZH"
    >
      <div
        className="sc-htoDjs eOdlka sc-gzVnrw iZIxNz"
      >
        Version
      </div>
      <div
        className="sc-gzVnrw iZIxNz"
      >
        1.0.0
      </div>
    </div>
    <div
      className="sc-bZQynM kzlsZH"
    >
      <div
        className="sc-htoDjs eOdlka sc-gzVnrw iZIxNz"
      >
        Developer
      </div>
      <div
        className="sc-gzVnrw iZIxNz"
      >
        Dominik Kundel
      </div>
    </div>
    <div
      className="sc-bZQynM kzlsZH"
    >
      <div
        className="sc-htoDjs eOdlka sc-gzVnrw iZIxNz"
      >
        Source Code
      </div>
      <div
        className="sc-gzVnrw iZIxNz"
      >
        <a
          href="https://github.com/dkundel/tunes"
          onClick={[Function]}
          target="_blank"
        >
          github.com/dkundel/tunes
        </a>
      </div>
    </div>
  </div>
  <button
    className="sc-dnqmqq dWsTFn"
    id="closeAbout"
    onClick={[Function]}
  >
    <i
      className="material-icons"
    >
      close
    </i>
  </button>
</div>
`;
